feat: Adds the ability have internal endpoints listen on ksql.internal.listener (#5212)
confluentinc/ksql: docs/operate-and-deploy/installation/server-config/config-reference.md

[ksqlDB Server image](https://hub.docker.com/r/confluentinc/ksqldb-server/)
is `KSQL_LISTENERS`.

### ksql.advertised.listener

> The `ksql.advertised.listener` setting controls the address advertised within

Reviewer: ```suggestion
The `ksql.advertised.listener` setting controls the address that is advertised within
```

> {{ site.kstreams }}, and shared with other ksqlDB nodes that make up the cluster.

Reviewer: ```suggestion
{{ site.kstreams }} and is shared with other ksqlDB nodes that make up the cluster.
```

This address is used to initiate intra-cluster requests such as forwarded pull queries,

> heartbeating, and lag reporting.

Author: Done

> If not set, the system will use the first of `listeners` to advertise and if this

Reviewer: ```suggestion
If not set, the system will use the first host in `listeners` to advertise, and if this
```

> is set to a wildcard address, then the system hostname is used.

Reviewer: ```suggestion
is set to a wildcard address, the system hostname is used.
```

> is set to a wildcard address, the system hostname is used.

Author: I made it behave this way and removed this explanation anyway, per your previous comment.

This setting is most often useful when other nodes in the cluster should use
a different endpoint from the one used in `listeners` to initiate a request
to a node in the cluster.

### ksql.internal.listener

The `ksql.internal.listener ` setting controls the address bound for use by internal,

> intra-cluster endpoints.  These include forwarded pull queries, heartbeating, and lag reporting.

Reviewer: ```suggestion
intra-cluster endpoints, which include forwarded pull queries, heartbeating, and lag reporting.
```

> If not set, the system will use `listeners` to expose internal endpoints.

Reviewer: ```suggestion
If not set, the system will use `listeners` to expose internal endpoints.
```
```suggestion
If not set, the system uses `listeners` to expose internal endpoints.
```

> This setting is most often useful in a IaaS environment to separate external-facing

Reviewer: ```suggestion
This setting is most often useful in an IaaS environment to separate external-facing
```
